### PR TITLE
[mle] make perferred partitionid config available for non reference device

### PR DESCRIPTION
**Pull request by simonlingoogle**
because it's used by ot-br-posix (openwrt).

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -1897,7 +1897,6 @@ otError Interpreter::ProcessPartitionId(uint8_t aArgsLength, char *aArgs[])
         OutputLine("%u", otThreadGetPartitionId(mInstance));
         error = OT_ERROR_NONE;
     }
-#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
```
Comment (zhanglongxia) [inline src/cli/cli.cpp:1900]: The  partitionid related macros`OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` are added by the [CL](https://github.com/openthread/openthread/pull/5872/files). Should we remove all macros `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` added by the [CL](https://github.com/openthread/openthread/pull/5872/files).

Comment (simonlingoogle) [reply src/cli/cli.cpp:1900]: readme fixed

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Comment by simonlingoogle** [comment]
Close for now because we decided to not add this partition setting feature to openwrt after discussed with @bukepo

**Pull request closed without merging**